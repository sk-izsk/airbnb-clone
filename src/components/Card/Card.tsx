import { Box, makeStyles, Typography } from '@material-ui/core';
import React from 'react';
import { CustomTheme } from '../../theme/muiTheme';

export interface CardProps {
  src?: string;
  title?: string;
  description?: string;
  price?: string;
}

const useStyles = makeStyles((theme: CustomTheme) => ({
  card: {
    margin: theme.spacing(1.25),
    borderRadius: 10,
    overflow: 'hidden',
    boxShadow: `0px 6px 18px -9px ${theme.palette.secondary.dark}`,
    transition: `transform 100ms ease-in`,
    cursor: 'pointer',
    '&:hover': {
      transform: 'scale(1.07)',
    },
  },
  cardInfo: {
    marginTop: -theme.spacing(1),
    borderRadius: 10,
    padding: theme.spacing(2.5),
    border: 1,
  },
  description: {
    fontSize: 14,
    fontWeight: 300,
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(1),
  },
  cardImg: {
    objectFit: 'fill',
    minWidth: theme.spacing(37.5),
    minHeight: theme.spacing(25),
    width: '100%',
  },
  title: {
    fontSize: 18,
    fontWeight: 600,
  },
}));

const Card: React.FC<CardProps> = ({ src, description, price, title }) => {
  const classes = useStyles();
  return (
    <Box className={classes.card}>
      <img className={classes.cardImg} src={src} alt='' />
      <Box className={classes.cardInfo}>
        <Typography className={classes.title} variant='h3'>
          {title}
        </Typography>
        <Typography className={classes.description} variant='body1'>
          {description}
        </Typography>
        <Typography variant='h5'>{price}</Typography>
      </Box>
    </Box>
  );
};

export { Card };
