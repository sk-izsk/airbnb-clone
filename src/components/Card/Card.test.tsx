import { render } from '@testing-library/react';
import React from 'react';
import { Card } from './Card';
const mockData = {
  src: 'mock src',
  description: 'mock desc',
  price: 'mock price',
  title: 'mock title',
};

test('Card renders correctly', () => {
  const { asFragment } = render(<Card {...mockData} />);
  expect(asFragment()).toMatchSnapshot();
});
