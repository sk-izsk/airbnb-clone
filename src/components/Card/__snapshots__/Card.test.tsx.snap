// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root MuiBox-root-6 makeStyles-card-1"
  >
    <img
      alt=""
      class="makeStyles-cardImg-4"
      src="mock src"
    />
    <div
      class="MuiBox-root MuiBox-root-7 makeStyles-cardInfo-2"
    >
      <h3
        class="MuiTypography-root makeStyles-title-5 MuiTypography-h3"
      >
        mock title
      </h3>
      <p
        class="MuiTypography-root makeStyles-description-3 MuiTypography-body1"
      >
        mock desc
      </p>
      <h5
        class="MuiTypography-root MuiTypography-h5"
      >
        mock price
      </h5>
    </div>
  </div>
</DocumentFragment>
`;
